test/interface/test_channel.rb

```diff
@@ -0,0 +1,189 @@
+
+                it "marshals an action model on send" do
+                    action_model = Actions::Models::Action.new("test")
+                    action_model.name = "action_model"
+                    @server.write_packet(action_model)
+                    ret = @client.read_packet
+                    assert_kind_of Protocol::ActionModel, ret
+                    assert_equal "action_model", ret.name
+                end
+
+                it "marshals an action argument on send" do
+                    action_model = Actions::Models::Action.new("test")
+                    action_model.name = "action_model"
+                    action_model.arguments <<
+                        Actions::Models::Action::Argument.new("arg")
+                    @server.write_packet(action_model)
+                    ret = @client.read_packet
+                    assert_kind_of Protocol::ActionModel, ret
+                    assert_equal "action_model", ret.name
+
+                    arg_out = ret.arguments.first
+                    assert_kind_of Protocol::ActionArgument, arg_out
+                    assert_equal "arg", arg_out.name
+                end
+
+                it "handles the Void null type from actions" do
+                    action_model = Actions::Models::Action.new("test")
+                    action_model.name = "action_model"
+                    action_model.arguments <<
+                        Actions::Models::Action::Argument.new(
+                            "arg", "bla", false, nil, Actions::Models::Action::Void
+                        )
+                    @server.write_packet(action_model)
+                    ret = @client.read_packet
+                    assert_kind_of Protocol::ActionModel, ret
+                    assert_equal "action_model", ret.name
+
+                    arg_out = ret.arguments.first
+                    assert_kind_of Protocol::ActionArgument, arg_out
+                    assert_equal "arg", arg_out.name
+                    assert_kind_of Protocol::VoidClass, arg_out.example
+                end
+
+                it "marshals an execution exception object" do
+                    plan.add(parent = Task.new)
+                    plan.add(child = Task.new)
+                    localized_error = Roby::LocalizedError.new(child)
+                    e = Roby::ExecutionException.new(localized_error)
+                    e.propagate(child, parent)
+
+                    @server.write_packet(e)
+                    ret = @client.read_packet
+                    assert_kind_of Protocol::ExecutionException, ret
+                    assert_equal "Roby::LocalizedError", ret.exception.class_name
+                    assert_equal child.droby_id.id, ret.failed_task.id
+                    assert_equal Set[parent.droby_id.id, child.droby_id.id],
+                                 ret.involved_tasks.map(&:id).to_set
+
+                end
+            end
+
+            describe "marshalling" do
+            end
```

Comment:
any tests you plan to add in this context? I see you already tested most marshaling features.

Reply:
Mmmmm ... actually I never did test the marshalling logic, only for specific classes. Just did. Thanks.